process/block/metablock.go

```diff
@@ -1506,6 +1496,10 @@ func (mp *metaProcessor) receivedShardHeader(headerHandler data.HeaderHandler, s
 		return
 	}
 
+	if !mp.blockTracker.ShouldAddHeader(shardHeader) {
```

Comment:
Actually you can remove this condition at all as it was used only to delete the out of range received headers. The conditions which protect to not request miniblocks in advance is right above.

Reply:
done